policyengine_us/variables/gov/states/ma/dta/tcap/tafdc/eligibility/non_financial/ma_tafdc_pregnancy_eligible.py

```diff
@@ -0,0 +1,30 @@
+from policyengine_us.model_api import *
+
+
+class ma_tafdc_pregnancy_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Eligible for Massachusetts Temporary Assistance for Families with Dependent Children (TAFDC) due to pregnancy"
+    definition_period = YEAR
+    reference = (
+        "https://www.law.cornell.edu/regulations/massachusetts/106-CMR-703-210"
+    )
+    defined_for = StateCode.MA
+
+    def formula(person, period, parameters):
+        is_pregnant = person("is_pregnant", period)
+        age = person("age", period)
+        current_pregnancy_month = person("current_pregnancy_month", period)
+        p = parameters(period).gov.states.ma.dta.tcap.tafdc.eligibility
+        teen_age_eligible = age < p.age_threshold.teen_parent
```

Comment:
can you refactor this to create an `age_limit` Python variable based on their student/parent/pregnancy status, then compare their age to that at the end?